### PR TITLE
Added ability to set proxy settings on OSX and Windows

### PR DESCRIPTION
**Pull request by opdude**

### CODE REVIEW

File: salt/states/proxy.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+# Import salt libs
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'proxy'
+
+
+def __virtual__():
+    '''
+    Only work on Mac OS
```
Comment (fracklen) [inline salt/states/proxy.py]: Seems counterintuitive to say "Only work on mac OS" and afterwards accept `__grains__['os'] == 'Windows']`

I'm confused...

Comment (fracklen) [inline salt/states/proxy.py]: Same here... I prefer the rebase to merging develop...

Comment (opdude) [inline salt/states/proxy.py]: Aye I would usually do that, but had already submitted the PR :)

**Comment by cachedout** [comment]
Thanks very much, @opdude!

**Comment by twangboy** [comment]
@opdude Hey, I know it's been a while since you've looked at this code. But, it looks like this sets proxy settings on Windows for the current user. Is this the intended functionality? Would it make more sense to set the proxy settings for everyone on the machine? Did you test this when salt is running as LocalSystem? Which `HKEY_CURRENT_USER` is being used?

**Comment by opdude** [comment]
@twangboy No we run salt as an elevated user at the moment so it's likely going to be that user that gets the proxy settings. I'll talk with some of my team members who are planning to move to such a scenario and see what they found out

**Pull request merged**